Origin: vendor
Bug-Debian: http://bugs.debian.org/615571
Bug: https://rt.cpan.org/Public/Bug/Display.html?id=65485
Forwarded: https://rt.cpan.org/Public/Bug/Display.html?id=65485
Last-Update: 2011-05-04

From f35f6e19700b9f0680484fd44c656ad971059c08 Mon Sep 17 00:00:00 2001
From: Niko Tyni <ntyni@debian.org>
Date: Wed, 4 May 2011 21:38:23 +0300
Subject: [PATCH] Use clone_sv() instead of clone_rv() for cloning references
 on newer perls

As of Perl 5.11, SVt_RV is gone and replaced by SVt_IV.

Using clone_sv() instead fixes test 5 of t/03scalar.t failing on 5.12.
---
 Fast.xs |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

--- a/Fast.xs
+++ b/Fast.xs
@@ -129,7 +129,11 @@
 #endif
 	(sv_clone_t)clone_sv,   // SVt_IV
 	(sv_clone_t)clone_sv,   // SVt_NV
+#if PERL_VERSION < 11
 	(sv_clone_t)clone_rv,   // SVt_RV
+#else
+	(sv_clone_t)clone_sv,   // SVt_IV
+#endif
 	(sv_clone_t)clone_sv,   // SVt_PV
 	(sv_clone_t)clone_sv,   // SVt_PVIV
 	(sv_clone_t)clone_sv,   // SVt_PVNV
